core/arch/emit_utils_shared.c

```diff
@@ -5534,6 +5536,56 @@ emit_special_ibl_xfer(dcontext_t *dcontext, byte *pc, generated_code_t *code, ui
         in = instrlist_first(custom_ilist);
     }
 
+#ifdef UNIX
+    /* i#4670: Jump to the unlinked IBL target if there are pending signals. This is
+     * required to bound delivery time for signals received while executing fragments
```

Comment:
I did 10 runs with and without  the fix of the proprietary app mentioned in i#4670. 



without-fix user seconds | with-fix user seconds
-- | --
109 | 264
230 | 297
246 | 297
308 | 301
316 | 322
329 | 323
334 | 351
338 | 358
354 | 369
383 | 439


There seem to be some variance between runs. So, removing first and last 10% as outliers, we get:
Average old vs new: 307 vs 327 seconds. So there's a 6.5% slowdown.

Reply:
Probably needs more runs for better confidence in the numbers.

Reply:
> I did 10 runs with and without the fix of the proprietary app mentioned in i#4670.

That app is doing heavyweight instrumentation which may mask other overhead?  Best to disable that and have just the drwrap portion.

Reply:
[drwrap-test-detach](https://github.com/DynamoRIO/dynamorio/blob/d28cd2d711059c00b5de6ec86f7618ce371b2bb1/suite/tests/client-interface/drwrap-test-detach.cpp#L129) seems suitable -- it loops on a function wrapped using `DRWRAP_RETURN_RETADDR`, and has only the wrap overhead, not other instrumentation.

- I noticed that the loop goes on for only [1ms](https://github.com/DynamoRIO/dynamorio/blob/d28cd2d711059c00b5de6ec86f7618ce371b2bb1/suite/tests/client-interface/drwrap-test-detach.cpp#L156), which leads to `pre_count`=`post_count`=0. So it'll be a good idea to increase this to 1s in the test suite. I have added a commit for that to this PR.
- For testing the overhead of this PR, I increased the sleep duration to 5 mins and compared `pre_count` from the before and after versions over 5 runs. Higher is faster.



without-fix pre_count | with-fix pre_count
-- | --
304008602 | 304555102
309676422 | 308248502
309824682 | 309160062
310135622 | 310072462
311061162 | 311737202

Harmonic mean of `pre_count` without fix: 308920630, with fix: 308735986
So there's a 0.06% slowdown after this PR.

Reply:
Also, I think I should've used the above methodology with the i#4670 proprietary app experiment too (in my earlier comment), because that app also sleeps for some time to allow threads to loop over some code. So overall execution time shouldn't have changed much before and after the change. Probably some other overhead dominated the execution time.

Reply:
OK, I think it would also be good to compare the tracing metrics on the large proprietary apps that were used when developing the replace-retaddr feature in the first place: we can talk offline about how to do that.